### PR TITLE
Plfm 2472

### PR DESCRIPTION
**Pull request by john-hill**
Please do not accept this yet.

### CODE REVIEW

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/principal/UserProfileUtillity.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import org.sagebionetworks.repo.model.principal.PrincipalAlias;
+
+
+/**
+ * Utilities for working with UserProfiles and PrincipalAliases.
+ * 
+ * @author John
+ *
+ */
+public class UserProfileUtillity {
```
Comment (jay-hodgson) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/principal/UserProfileUtillity.java:21]: rename UserProfileUtillity to UserProfileUtility, isTempoaryUsername to isTemporaryUsername, and createTempoaryUserName to createTemporaryUserName, and groupAlieaseByPrincipal to groupAliasesByPrincipal

File: lib/models/src/main/java/org/sagebionetworks/repo/model/UserInfo.java [hunk 2]
```diff
@@ -51,17 +56,43 @@ public void setIndividualGroup(UserGroup individualGroup) {
...
+
+	public Long getId() {
+		return id;
+	}
+
+	public void setId(Long id) {
+		this.id = id;
+	}
+
+	public String getEtag() {
```
Comment (jay-hodgson) [inline lib/models/src/main/java/org/sagebionetworks/repo/model/UserInfo.java]: remove etag

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/UserProfileUtils.java [hunk 3]
```diff
@@ -117,4 +117,13 @@ public static Favorite copyDboToDto(DBOFavorite dbo) throws DatastoreException {
 		return dto;
 	}
 
+	public static DBOUserProfile translateFromStack26Form(DBOUserProfile backup) {
```
Comment (jay-hodgson) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/UserProfileUtils.java]: clean up

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/principal/PrincipalAliasDaoImpl.java [hunk 4]
```diff
@@ -30,15 +39,25 @@
  */
 public class PrincipalAliasDaoImpl implements PrincipalAliasDAO {
 	
+	private static final String SQL_LOCK_PRINCIPAL = "SELECT "+COL_USER_GROUP_ID+" FROM "+TABLE_USER_GROUP+" WHERE "+COL_USER_GROUP_ID+" = ? FOR UPDATE";
+	private static final String SQL_DELETE_ALL_ALIASES_FOR_PRINCIPAL = "DELETE FROM "+TABLE_PRINCIPAL_ALIAS+" WHERE "+COL_PRINCIPAL_ALIAS_PRINCIPAL_ID+" = ?";
```
Comment (jay-hodgson) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/principal/PrincipalAliasDaoImpl.java:43]: clean up?

**Comment by jay-hodgson** [comment]
need to think about email validation (with multiple addresses)

PLFM-2486

File: services/repository-managers/src/main/java/org/sagebionetworks/bridge/manager/participantdata/ParticipantDataManagerImpl.java [hunk 5]
```diff
@@ -51,7 +56,9 @@ public PaginatedRowSet getData(UserInfo userInfo, String participantDataId, Inte
 		List<String> participantIds = participantDataMappingManager.mapSynapseUserToParticipantIds(userInfo);
 		String participantId = participantDataDAO.findParticipantForParticipantData(participantIds, participantDataId);
 		if (participantId == null) {
-			throw new NotFoundException("No data found for this user");
+			// User has never created data for this ParticipantData type, which is not an error, so return
+			// empty result. It will have no headers given the way this works.
+			return getEmptyPaginatedRowSet();
```
Comment (jay-hodgson) [inline services/repository-managers/src/main/java/org/sagebionetworks/bridge/manager/participantdata/ParticipantDataManagerImpl.java:61]: need to do this?

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/MessageManagerImpl.java [hunk 6]
```diff
@@ -645,30 +648,31 @@ public void sendPasswordResetEmail(String recipientId, DomainType originClient,
...
+		UserInfo recipient = userManager.getUserInfo(recipientId);
 		String domain = WordUtils.capitalizeFully(originClient.name());
 		String subject = "Welcome to " + domain + "!";
 		String messageBody = readMailTemplate("message/WelcomeTemplate.txt");
 		messageBody = messageBody.replaceAll(TEMPLATE_KEY_ORIGIN_CLIENT, domain);
 		
 		//TODO use the Alias here
-		String alias = recipient.getUser().getDisplayName();
+		UserProfile profile = this.userProfileDAO.get(recipientId.toString());
+		String alias = profile.getDisplayName();
```
Comment (jay-hodgson) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/MessageManagerImpl.java]: I've been generally avoiding displayname field, in favor of using the first name and last name fields.  
But in this case (and below, on line 689), shouldn't it be using getUserName()?

Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/MessageManagerImpl.java]: Yes

File: services/repository/src/main/java/org/sagebionetworks/repo/web/service/UserProfileServiceImpl.java [hunk 7]
```diff
@@ -79,13 +86,13 @@
 	 */
 	private volatile Long cachesLastUpdated = 0L;
 	private volatile Trie<String, Collection<UserGroupHeader>> userGroupHeadersNamePrefixCache;
```
Comment (jay-hodgson) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/service/UserProfileServiceImpl.java:88]: Are you planning to replace the Trie for team search later?

**Pull request merged**